### PR TITLE
Fix Scroll Lock

### PR DESCRIPTION
**Pull request by jehoshua02**
http://jsbin.com/dapupo/16/edit?css,js,output

Copied in markup to example page.
Copied in javascript and styles.
Fix ensureOverlay to wait for document ready.

### CODE REVIEW

**Comment by jehoshua02** [comment]
Fixes: #6, #7, #9, #10, #11, #12, #13

**Comment by jehoshua02** [comment]
I started to draft [release notes] for this as v1.0.0.

**Comment by jehoshua02** [comment]
@kbratko If you please, feel free to review this and give a :+1:. Thanks.

**Comment by kbratko** [comment]
:+1:

File: example/menu.js [hunk 1]
```diff
@@ -1,236 +1,182 @@
...
+      ], function (event) {
+        return event + '.scroll-kill' + eventNamespace;
+      }).join(' ');
+
+      $(el).on(events, function (event) {
+        event.preventDefault();
+        event.stopPropagation();
+      });
+    },
+    stuff: function (el) {
```
Comment (janiv) [inline example/menu.js]: What do you think of hasStuff or canScroll?

File: src/menu.js [hunk 2]
```diff
@@ -1,236 +1,182 @@
...
+      ], function (event) {
+        return event + '.scroll-kill' + eventNamespace;
+      }).join(' ');
+
+      $(el).on(events, function (event) {
+        event.preventDefault();
+        event.stopPropagation();
+      });
+    },
+    stuff: function (el) {
```
Comment (janiv) [inline src/menu.js]: What do you think of hasStuff or canScroll?

**Comment by janiv** [comment]
Looks a lot better. Awesome work.

Here's the problems I saw:

* Open the menu and try to use keyboard scrolling (spacebar or arrow keys). The main content area scrolls instead of the menu content area.
* For accessibility, I wonder if the menu open toggles should be button tags instead of i tags. This would allow for tabbing to the menu open icons and better keyboard interactions. This could probably be another story.
* You'll love this one. It's almost impossible to open the menu on an iPhone 6+ in landscape mode in Safari. When it did open, the animations were very choppy.
* We should make it so pushing esc closes the open menus. Just a random thought during the review and this is not needed for this story to be completed. 

Finally, here's how it looked in IE8. The menu opened if I pushed the button but I could never get the menu icons to show or work.

screen shot 2015-04-01 at 2 44 38 pm

File: src/menu.js (hunk 2 above)
Comment (jehoshua02) [inline src/menu.js]: I like it.

Comment (jehoshua02) [inline src/menu.js]: Commit pushed.

**Comment by jehoshua02** [comment]
@janiv 

> Open the menu and try to use keyboard scrolling (spacebar or arrow keys). The main content area scrolls instead of the menu content area.

Good stuff. Issue entered (#15).

> For accessibility, I wonder if the menu open toggles should be button tags instead of i tags. This would allow for tabbing to the menu open icons and better keyboard interactions. This could probably be another story.

Yeah maybe. But that is a problem with the example page and not ddm-menu itself.

> You'll love this one. It's almost impossible to open the menu on an iPhone 6+ in landscape mode in Safari. When it did open, the animations were very choppy.

That's two issues: 1) hard to open 2) choppy animations.

I made the menu toggles click area larger. I don't know if I did that before moving from jsbin into ddm-menu or after.

Choppy animations: I'll wait for product owners to find a problem with that.

> We should make it so pushing esc closes the open menus. Just a random thought during the review and this is not needed for this story to be completed.

I would love to reach the zenith of my career by mastering accessibility, on top of building tdd, responsive, cross-browser, react applications. I put in an issue (#16).

> Finally, here's how it looked in IE8. The menu opened if I pushed the button but I could never get the menu icons to show or work.

You broke it! It's all your fault. I'm testing locally on http://projects.ksl.com/deseretdigital-ui/ddm-menu/example/ and it looks fine.

I had to downgrade to jQuery 1 to get it to work though. I've updated the bower.json. Until we official drop support for IE8.

screen shot 2015-04-01 at 3 53 22 pm

screen shot 2015-04-01 at 3 53 31 pm

**Comment by jehoshua02** [comment]
@janiv GIMME A THUMB!

**Comment by janiv** [comment]
:+1: :star: :wheelchair: :star: :+1:

**Comment by jehoshua02** [comment]
@janiv Okay, I'm going to merge and release! Thanks!

**Pull request merged**